src/databricks/labs/ucx/azure/resources.py

```diff
@@ -117,44 +121,134 @@ def inner() -> dict[str, str]:
 
         return _credentials
 
+    def get(self, path: str, host: str, api_version: str | None = None):
+        headers = {"Accept": "application/json"}
+        query = {}
+        if api_version is not None:
+            query = {"api-version": api_version}
+        if host == "azure_mgmt":
+            return self._resource_manager.do("GET", path, query, headers)
+        if host == "azure_graph":
+            return self._graph.do("GET", path)
+        return None
+
+    def put(self, path: str, host: str, body: dict | None = None):
+        headers = {"Content-Type": "application/json"}
+        if host == "azure_mgmt":
+            assert body is not None
+            query = {"api-version": "2022-04-01"}
+            return self._resource_manager.do("PUT", path, headers=headers, body=body, query=query)
+        if host == "azure_graph":
+            if body is not None:
+                return self._graph.do("POST", path, headers=headers, body=body)
+            return self._graph.do("POST", path, headers=headers)
+        raise ValueError()
+
+    @property
+    def token(self):
```

Comment:
Why do we need this as the public method? Remove.

Reply:
this is called in the AzureResources to get the token and retrieve tenantid